packages/language/core/v1/src/inputs.ts

```diff
@@ -115,23 +116,49 @@ export type NormalInput = InputContent;
  */
 export const NormalInputGuard = InputContentGuard;
 
+export interface MerkleizedHashAndContinuation {
+  continuation_hash: BuiltinByteString;
+  merkleized_continuation: Contract;
+}
+
+export const MerkleizedHashAndContinuationGuard: t.Type<MerkleizedHashAndContinuation> = t.type({
+  continuation_hash: BuiltinByteStringGuard,
+  merkleized_continuation: ContractGuard,
+});
+
+export type MerkleizedDeposit = IDeposit & MerkleizedHashAndContinuation;
+
+export const MerkleizedDepositGuard: t.Type<MerkleizedDeposit> = t.intersection([
+  IDepositGuard,
+  MerkleizedHashAndContinuationGuard,
+]);
+
+export type MerkleizedChoice = IChoice & MerkleizedHashAndContinuation;
+
+export const MerkleizedChoiceGuard: t.Type<MerkleizedChoice> = t.intersection([
+  IChoiceGuard,
+  MerkleizedHashAndContinuationGuard,
+]);
+
+// NOTE: Because INotify is serialized as a string, it is invalid to do the &.
+//       the type in marlowe-cardano is serialized just as the hash and continuation.
+export type MerkleizedNotify = MerkleizedHashAndContinuation;
+export const MerkleizedNotifyGuard = MerkleizedHashAndContinuationGuard;
+
 /**
  * TODO: Revisit
  * @category Input
  */
-export type MerkleizedInput = t.TypeOf<typeof MerkleizedInputGuard>;
+export type MerkleizedInput = MerkleizedDeposit | MerkleizedChoice | MerkleizedNotify;
 /**
  * TODO: Revisit
  * @category Input
  */
-export const MerkleizedInputGuard = t.intersection([
-  InputContentGuard,
-  t.type({
-    continuation_hash: BuiltinByteStringGuard,
-    merkleized_continuation: ContractGuard,
-  }),
+export const MerkleizedInputGuard =t.union([
```

Comment:
Is this formatted?

Reply:
maybe not, will try to re-run the formatting after last fixes